engine/src/flutter/lib/web_ui/lib/src/engine/canvaskit/picture.dart

```diff
@@ -96,30 +97,47 @@ class CkPicture implements LayerPicture {
   }
 
   @override
-  CkImage toImageSync(int width, int height) {
-    assert(debugCheckNotDisposed('Cannot convert picture to image.'));
-
+  ui.Image toImageSync(int width, int height) {
+    assert(!debugDisposed);
     final Surface surface = CanvasKitRenderer.instance.pictureToImageSurface;
-    final CkSurface ckSurface = surface.createOrUpdateSurface(BitmapSize(width, height));
-    final CkCanvas ckCanvas = ckSurface.getCanvas();
+    surface.setSize(BitmapSize(width, height));
+    final CkSurface ckSurface = surface as CkSurface;
+    final SkSurface skiaSurface = ckSurface.skSurface!;
+
+    final CkCanvas ckCanvas = CkCanvas.fromSkCanvas(skiaSurface.getCanvas());
     ckCanvas.clear(const ui.Color(0x00000000));
     ckCanvas.drawPicture(this);
-    final SkImage skImage = ckSurface.surface.makeImageSnapshot();
-    final SkImageInfo imageInfo = SkImageInfo(
-      alphaType: canvasKit.AlphaType.Premul,
-      colorType: canvasKit.ColorType.RGBA_8888,
-      colorSpace: SkColorSpaceSRGB,
-      width: width.toDouble(),
-      height: height.toDouble(),
+    final SkImage skImage = skiaSurface.makeImageSnapshot();
+
+    // TODO(hterkelsen): This is a hack to get the pixels from the SkImage.
+    // We should be able to do this without creating a new image. This is
+    // a workaround for a bug in CanvasKit.
```

Comment:
Do we have a link to the bug?

Reply:
I think I filed one with the Skia team a while back. I'll try to find it